internal/controller/promotion/helm.go

```diff
@@ -90,16 +94,19 @@ func (h *helmer) apply(
 			update.Helm.Charts,
 		)
 	if err != nil {
-		return nil, fmt.Errorf("error preparing changes to affected Chart.yaml files: %w", err)
+		return nil, fmt.Errorf("preparing changes to affected Chart.yaml files: %w", err)
 	}
 	for chart, changes := range changesByChart {
 		chartPath := filepath.Join(workingDir, chart)
 		chartYAMLPath := filepath.Join(chartPath, "Chart.yaml")
 		if err = h.setStringsInYAMLFileFn(chartYAMLPath, changes); err != nil {
-			return nil, fmt.Errorf("error updating dependencies for chart %q: %w", chart, err)
+			return nil, fmt.Errorf("setting dependency versions for chart %q: %w", chart, err)
+		}
+		if err = h.prepareDependencyCredentialsFn(context.TODO(), homeDir, chartYAMLPath, namespace); err != nil {
```

Comment:
Other than needing to tweak some function signatures and function calls, is there anything preventing us from getting at the right context here?

Reply:
Not really, but the size of the constructor is starting to feel outrageous and a mixed bag of not-always-so-relevant things.